### PR TITLE
Request to add constructors for parts of a ship

### LINKED ISSUE #16: Parts of a ship
Parts of a ship it is what you can buy for your ship to improve
- [ ]  Hull 
  * should include gun
- [ ]  Engine
  * Must have engine power

### PR DESCRIPTION
**Pull request by ghost**
Added constructors for parts of a ship bd3c3c7
- Added constructor for class ShipPart
- Added constructor for class Engine
- Added constructor for class Hull

Resolve #16

### CODE REVIEW

**Comment by ghost** [comment]
Fixed some bugs, please, reviewed again.

File: src/main/java/rape/brutal/gitstarship/parts/Engine.java [hunk 1]
```diff
@@ -10,12 +10,13 @@
  */
 public class Engine extends ShipPart {
 
-    private int cost;
-    private String name;
-    private int size;
-
     private int power;
 
+    public Engine() {
```
Comment (ViRGiL175) [inline src/main/java/rape/brutal/gitstarship/parts/Engine.java]: Power parameter is needed.

You can do it via parameter or via concrete subclasses: `SmallEngine`, `MediumEngine`, `ExtraEngine`.

File: src/main/java/rape/brutal/gitstarship/parts/Hull.java [hunk 2]
```diff
@@ -7,31 +7,57 @@
...
  */
 public class Hull extends ShipPart {
 
-    private int cost;
-    private String name;
-    private int size;
+    private Gun[] guns;
 
-    private Gun[] guns = new Gun[8];
+    public Hull() {
```
Comment (ViRGiL175) [inline src/main/java/rape/brutal/gitstarship/parts/Hull.java]: Maybe different kinds of Hulls with different amount of gun slots?

**Review by ViRGiL175 — CHANGES_REQUESTED** [review]

**Comment by ViRGiL175** [comment]
Reviewed.

File: src/main/java/rape/brutal/gitstarship/parts/Engine.java (hunk 1 above)
Comment (ghost) [reply src/main/java/rape/brutal/gitstarship/parts/Engine.java]: Fixed!

**Comment by ghost** [comment]
Fixed some bugs, please, reviewed again!

**Review by ViRGiL175 — APPROVED** [review]
It can be better.

**Comment by ViRGiL175** [comment]
Approved.

**Pull request merged**